### PR TITLE
refactor mocked sandbox specs

### PR DESCRIPTION
**Pull request by scottbartell**

### CODE REVIEW

File: spec/mocked_sandbox/auth_capture_spec.rb [hunk 1]
```diff
@@ -1,109 +1,70 @@
...
+      order_id: "12345",
       expiry_month: card.expiry_month,
       expiry_year: card.expiry_year
-    )
+    ).tap do
+      Vantiv::MockedSandbox.disable_self_mocked_requests!
+    end
   end
 
+  let(:mocked_response) { run_mocked_response }
```
Comment (ChrisVoxland) [inline spec/mocked_sandbox/auth_capture_spec.rb]: why not just rename `#run_mocked_response` to mocked response?

File: spec/mocked_sandbox/auth_capture_spec.rb [hunk 2]
```diff
@@ -1,109 +1,70 @@
...
+
   after { Vantiv::MockedSandbox.disable_self_mocked_requests! }
```
Comment (ChrisVoxland) [inline spec/mocked_sandbox/auth_capture_spec.rb]: is this still needed?

File: spec/mocked_sandbox/auth_spec.rb [hunk 3]
```diff
@@ -1,77 +1,54 @@
...
       expiry_year: card.expiry_year
-    )
+    ).tap do
+      Vantiv::MockedSandbox.disable_self_mocked_requests!
+    end
   end
 
+  let(:mocked_response) { run_mocked_response }
+
   after { Vantiv::MockedSandbox.disable_self_mocked_requests! }
```
Comment (ChrisVoxland) [inline spec/mocked_sandbox/auth_spec.rb]: still needed?

File: spec/mocked_sandbox/capture_spec.rb [hunk 4]
```diff
@@ -1,83 +1,54 @@
...
+      transaction_id: transaction_id,
+      amount: 10100
+    ).tap do
+      Vantiv::MockedSandbox.disable_self_mocked_requests!
+    end
   end
 
+  let(:mocked_response) { run_mocked_response }
+
   after { Vantiv::MockedSandbox.disable_self_mocked_requests! }
```
Comment (ChrisVoxland) [inline spec/mocked_sandbox/capture_spec.rb]: needed?

File: spec/mocked_sandbox/credit_spec.rb [hunk 5]
```diff
@@ -1,84 +1,54 @@
...
+      transaction_id: transaction_id,
+      amount: 10100
+    ).tap do
+      Vantiv::MockedSandbox.disable_self_mocked_requests!
+    end
   end
 
+  let(:mocked_response) { run_mocked_response }
+
   after { Vantiv::MockedSandbox.disable_self_mocked_requests! }
```
Comment (ChrisVoxland) [inline spec/mocked_sandbox/credit_spec.rb]: same

File: spec/mocked_sandbox/tokenize_by_direct_post_spec.rb [hunk 6]
```diff
@@ -1,65 +1,50 @@
...
       expiry_year: card.expiry_year,
       cvv: card.cvv
     )
+    Vantiv::MockedSandbox.disable_self_mocked_requests!
+    response
   end
 
+  let(:mocked_response) { run_mocked_response }
+
   after { Vantiv::MockedSandbox.disable_self_mocked_requests! }
```
Comment (ChrisVoxland) [inline spec/mocked_sandbox/tokenize_by_direct_post_spec.rb]: needed?

File: spec/mocked_sandbox/tokenize_by_direct_post_spec.rb [hunk 7]
```diff
@@ -1,65 +1,50 @@
...
-    Vantiv.tokenize_by_direct_post(
+  def run_mocked_response
+    Vantiv::MockedSandbox.enable_self_mocked_requests!
+    response = Vantiv.tokenize_by_direct_post(
       card_number: card.card_number,
       expiry_month: card.expiry_month,
       expiry_year: card.expiry_year,
       cvv: card.cvv
     )
+    Vantiv::MockedSandbox.disable_self_mocked_requests!
```
Comment (ChrisVoxland) [inline spec/mocked_sandbox/tokenize_by_direct_post_spec.rb]: why not use the same tap style you have been here?

File: lib/vantiv/api.rb [hunk 8]
```diff
@@ -5,6 +5,12 @@
 require 'vantiv/api/account_updater_response'
 require 'vantiv/api/tokenization_response'
 
+require 'vantiv/api/card'
+require 'vantiv/api/transaction'
+require 'vantiv/api/payment_account'
+require 'vantiv/api/address'
+require 'vantiv/api/fraud_result'
```
Comment (aarondcohen) [inline lib/vantiv/api.rb]: why not alphabetize?

File: spec/mocked_sandbox/auth_capture_spec.rb [hunk 9]
```diff
@@ -1,109 +1,68 @@
...
-
-      it "returns a raw body string" do
+      it "returns the same attributes in the live and mocked responses" do
+        expect(live_response.success?).to eq mocked_response.success?
+        expect(live_response.failure?).to eq mocked_response.failure?
+        expect(live_response.message).to eq mocked_response.message
+        expect(live_response.error_message).to eq mocked_response.error_message
+        expect(live_response.httpok).to eq mocked_response.httpok
+        expect(live_response.http_response_code).to eq mocked_response.http_response_code
+        expect(live_response.api_level_failure?).to eq mocked_response.api_level_failure?
```
Comment (aarondcohen) [inline spec/mocked_sandbox/auth_capture_spec.rb]: this is why loops exist:
```ruby
%i( success? failure? ... ).each do |attribute|
  expect(live_response.send(attribute)).to eq(mocked_response.send(attribute))
end
```

File: spec/mocked_sandbox/auth_reversal_spec.rb [hunk 10]
```diff
@@ -1,81 +1,48 @@
...
-
-      it "returns a raw body string" do
+      it "returns the same attributes in the live and mocked responses" do
+        expect(live_response.success?).to eq mocked_response.success?
+        expect(live_response.failure?).to eq mocked_response.failure?
+        expect(live_response.message).to eq mocked_response.message
+        expect(live_response.error_message).to eq mocked_response.error_message
+        expect(live_response.httpok).to eq mocked_response.httpok
+        expect(live_response.http_response_code).to eq mocked_response.http_response_code
+        expect(live_response.api_level_failure?).to eq mocked_response.api_level_failure?
```
Comment (aarondcohen) [inline spec/mocked_sandbox/auth_reversal_spec.rb]: I'm starting to notice a pattern here... maybe a function that extracts the attributes into a hash?  then you can compare the 2 hashes and write the extraction code once.

File: spec/mocked_sandbox/auth_capture_spec.rb (hunk 9 above)
Comment (scottbartell) [inline spec/mocked_sandbox/auth_capture_spec.rb]: I just basically refactored it from that. I really dislike iterating through a series a methods to send in specs - I find that it leads to complicated failure messages and ultimately slows down development.

**Comment by scottbartell** [comment]
I think this is a significant improvement so I'm going to merge this and possibility address the spec duplication issues at a later time.

**Pull request merged**